Digipost/OAuthToken+IDToken.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  OAuthToken+IDToken.swift
+//  Digipost
+//
+//  Created by Håkon Bogen on 19/05/15.
+//  Copyright (c) 2015 Posten. All rights reserved.
+//
+
+import UIKit
+
+private struct OAuthTokenIdTokenConstants {
+    static let nonce = "nonce"
+    static let aud = "aud"
+}
+
+extension OAuthToken {
+
+    class func isIdTokenValid(idToken : String?, nonce : String) -> Bool {
+        if let actualIDToken = idToken {
+            let idTokenContentArray  = idToken?.componentsSeparatedByString(".")
+            if idTokenContentArray?.count == 2 {
+                if let base64EncodedJson = idTokenContentArray?[1] {
+                    var numberOfCharactersAdded = 0
+                    var error : NSError?
+                    var alteredBase64EncodedJson = base64EncodedJson
+                    var base64Data : NSData?
+                    while base64Data == nil {
```

Comment:
As i remember the backend delivered base64 that wouldn't be recognized by the app, so it had to add trailing =, probably not as many as 7, but from 0 to 2 i guess.